firmware-support/gdb-trace/src/lib.rs

```diff
@@ -0,0 +1,29 @@
+// SPDX-FileCopyrightText: 2022 Google LLC
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#![no_std]
+
+use core::{fmt::Write, panic::PanicInfo};
+
+/// Internal function for a GDB panic handler. It disables interrupts and attemps to write
+/// the panic info to the provided write-able interface. It's expected that the user will
+/// be running a GDB process and set a breakpoint to the panic handler so that the output
+/// can be read.
```

Comment:
Will we always have GDB though? Couldn't we _also_ write to UART here?

Reply:
Right, yes, it could just do that. That's totally an option - nothing in this function _requires_ that you have GDB hook a breakpoint on this function, it's just there for convenience so that there's a fixed name to use. It takes some generic writer in order to enable that.

Reply:
Ahhh okay, I get how to read this now. So in the current configuration nothing really changes. If we _don't_ trap with GDB it'll happily print to UART, because we pass that as `$writer`?

Reply:
Yeah, if you define your panic handler using the provided convenience macro. If you need something else then the `gdb_panic_internal` function is there to handle writing the panic info to whatever is provided, even if that's a null writer.